transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/util/Util.java

```diff
@@ -111,14 +109,19 @@ public static boolean draggingFromValidTarget(TransformationManager manager) {
     }
 
     public static void ensureSourceFolderExists(IJavaProject javaProject,
-                                                String folder,
-                                                IProgressMonitor monitor) throws JavaModelException {
+                                                String folderName,
+                                                IProgressMonitor monitor) throws Exception {
         boolean exists = false;
         IClasspathEntry[] entries = javaProject.getRawClasspath();
-        IPath path = javaProject.getPath().append(folder);
+        IPath path = javaProject.getPath().append(folderName);
         for (IClasspathEntry entry : entries) {
             if (entry.getEntryKind() == IClasspathEntry.CPE_SOURCE && entry.getPath().equals(path)) {
                 exists = true;
+                File folder = javaProject.getResource().getLocation().append(folderName).toFile();
+                if (!folder.exists()) {
+                    if (!folder.mkdirs()) throw new Exception(Messages.bind(Messages.Util_UnableToCreateSourceFolder, folder));
```

Comment:
It might be best to use an IOException instead of the generic Exception
We could throws IOException and CoreException on the method.